camviewer_ui_impl.py

```diff
@@ -694,7 +706,28 @@ def __init__(
             self.onCameraSelect(int(cameraIndex))
         except Exception:
             pass
+
+        # Set the right hand area's width based on font sizes
+        font = self.ui.labelCamera.font()
+        fm = QFontMetricsF(font)
+        charwidth = fm.averageCharWidth()
+        # We want to make it wide enough for 50 average chars
+        width = 50 * charwidth
+        self.ui.rightScrollArea.setFixedWidth(width)
+        self.ui.rightPanelWidget.setFixedWidth(width)
+
         self.efilter = FilterObject(self.app, self)
+
+        # Timer with 0 delay = end of event queue
+        # i.e. immediately after all the queued initial rendering
+        self.late_init_timer = QTimer(self)
+        self.late_init_timer.singleShot(0, self.late_init)
+
+    def late_init(self):
+        """
+        Init routines to be done immediately after the initial render
+        """
+        # Avoid weird sizing race conditions from doing this in __init__
```

Comment:
sigh qt really does end up being a bunch of spaghetti if we let it huh... 

I guess the problems arise from setting this filter then having the scroll area size change throughout init?

Reply:
This is worse or stranger than anything you were expecting

When I did this during init, I saw the following sequence at startup:
- show event filter starts
- hide event filter starts
- hide event filter ends
- show event filter ends

So even though the sequence was show first, then hide, so we should have the smaller of the two width values, we ended with the show width value because the event handler executions were interleaved...

Delaying till later skips these first few events without writing a janky "skip your first event ok please" sort of deal into the event filter, which seemed like it could be fragile.

Reply:
I agree with the approach taken here, I'd rather apply the event filter late than have to track the ins-and-outs of the initial event sequence